src/stateChangeCapture.ts

```diff
@@ -8,29 +8,36 @@ export interface LastItemState {
 }
 
 export function sanitizePipelineName(pipelineName: string | undefined): string {
-
   const sanitisePatterns = process.env.SANITISE_PATTERNS || "-pipeline"
   const removeWhitespace = (x: string) => x.trim()
 
   if (pipelineName) {
     const sanitise = pattern => {
-      const matchResult = pipelineName.split(pattern)
-      if (matchResult.length > 1) {
-        return matchResult[0]
-      } else {
-        //no match
-        return undefined
+      if (process.env.IS_REGEX && process.env.IS_REGEX == "true") {
+        const matchResult = pipelineName.match(pattern)
+        if (matchResult === null) {
+          return undefined
+        } else {
+          return pattern.includes('(.*)') ?  matchResult[1] : matchResult[0] }
```

Comment:
You're checking for a specific capture group pattern here.

I don't think that's acceptable.  If we support regex, then we should support regex.  Perhaps it needs more thought.

The following are all valid regex:
```
/(.*)-pipeline-(.*)/
/(prefix)-pipeline-(.*)/
/(foo)-ignored-([abc]+)/
```

I could come up with more examples.

Reply:
Agreed. I should potentially add more examples for this.

Should i concatenate all the matches to give the sanitized pattern? Is that what we want to do with the examples above?

Or should i just return the first match when multiple things on the regex match the pipeline names.